analysis/type.go

```diff
@@ -99,6 +100,56 @@ func (a *DefaultAnalyzer) Analyze(input []byte) TokenStream {
 
 var ErrInvalidDateTime = fmt.Errorf("unable to parse datetime with any of the layouts")
 
+const UnixSecs = "unix_sec"
+const UnixMilliSecs = "unix_milli"
+const UnixMicroSecs = "unix_micro"
+const UnixNanoSecs = "unix_nano"
+
+type TimestampBounds struct {
+	Min int64
+	Max int64
+}
+
+var UnixTimestampFormats = map[string]*TimestampBounds{
```

Comment:
i think its better to not use pointer/reference type over here, mainly since these formats are supposed to be constant and overriding of the values within these formats should be forbidden.